torch/onnx/utils.py

```diff
@@ -778,9 +778,24 @@ def set_names(node_list, name_list, descriptor):
             raise RuntimeError(
                 "number of %s names provided (%d) exceeded number of %ss (%d)"
                 % (descriptor, len(name_list), descriptor, len(node_list)))
-        for name, node in zip(name_list, node_list):
+
+        # Mark if the node DebugName is set before.
+        node_set = set()
+        for i, (name, node) in enumerate(zip(name_list, node_list)):
+            if node in node_set:
+                # Duplicated node, insert onnx::Identity to avoid the same DebugName.
+                identity_node = graph.create("onnx::Identity")
+                identity_node.insertAfter(node.node())
+                identity_node.addInput(node)
+                identity_node.output().setType(node.type())
+                graph.return_node().replaceInput(i, identity_node.output())
```

Comment:
This part of logic is exclusively for graph outputs. However function `set_names` is also called on graph inputs. Could you refactor the code a bit to make it more robust that setting graph input names don't go through this route? Otherwise we are depending on a hidden assumption that there won't be duplicates for graph inputs.

Reply:
Sure, the graph inputs is indeed considerd into this function.

Reply:
It looks like that the graph `input` nodes won't be duplicated. Just as the latest test I updated shown. Even I put the some input `(x, x)`, the node is also different.

Reply:
What if users by mistake provide the same name (duplicate names) for two inputs?
Regardless, I also think it's better to refactor this logic for outputs only.

Reply:
> What if users by mistake provide the same name (duplicate names) for two inputs?
> Regardless, I also think it's better to refactor this logic for outputs only.

Only the two same nodes will go through this route.
You are right, I add the logic for outputs node. Thanks for it!